.github/workflows/reusable-build-apk.yml

```diff
@@ -0,0 +1,139 @@
+name: "[Reusable] Build APK"
+
+on:
+    workflow_call:
+        inputs:
+            build-environment:
+                type: string
+                required: true
+        outputs:
+            build-artifact:
+                description: "The artifact file name of the built APK"
+                value: ${{ jobs.build-apk.outputs.build-artifact }}
+            optional-mapping-file:
+                description: "The mapping file for the built APK"
+                value: ${{ jobs.build-apk.outputs.mapping-text }}
+
+jobs:
+    build-apk:
+
+        outputs:
+            build-artifact: ${{ steps.set-build-artifact.outputs.artifact }}
+            mapping-text: ${{ steps.build-internal-apk.outputs.mapping }}
+
+        runs-on: ubuntu-latest
+
+        environment: ${{ inputs.build-environment }}
+
+        env:
+            # VERSION_CODE: Unique version code using the sum of the current timestamp and run number.
+            VERSION_CODE: "set in lower step"
+            # VERSION_NAME: Version name derived from the GitHub ref name after the final /.
+            VERSION_NAME: "set in lower step"
+            # VERSION_SUFFIX: The build environment (e.g., dev, staging, internal).
+            VERSION_SUFFIX: ${{ inputs.build-environment }}
+            # VERSION_BUILD: A unique build identifier combining the run number and attempt.
+            VERSION_BUILD: ${{ github.run_number }}.${{ github.run_attempt }}
+            # The final output file name.
+            FILE_NAME: "set in lower step"
+            # The base floor version code. Please check the README for more information.
+            BASE_VERSION_CODE: 10000000
+
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v4
+
+            -   name: Set up JDK 17
+                uses: actions/setup-java@v4
+                with:
+                    java-version: 17
+                    distribution: 'temurin'
+
+            -   name: Check workflow validity
+                if: ${{ github.run_attempt > 99 }}
+                run: |
+                    echo "Run attempts exceeded 99. Please start a new workflow."
+                    exit 1
+
+            -   name: Check base version code
+                if: ${{ github.run_number * 100 >= env.BASE_VERSION_CODE }}
+                run: |
+                    echo "Github workflows now exceeds the base version code. Please remove the BASE_VERSION_CODE"
+                    echo "BASE_VERSION_CODE=0" >> $GITHUB_ENV
+
+            -   name: Compute version variables
+                run: |
+                    echo "VERSION_CODE=$(($BASE_VERSION_CODE + ${{ github.run_number }} * 100 + ${{ github.run_attempt }}))" >> $GITHUB_ENV
```

Comment:
Hm, I'm confused - is it going to be `run_number*100+run_attempt` or `run_number`.`run_attempt`?
Ie. if run_number is 123 and run_attempt is 45:
`12345` vs `123.45`

Reply:
The version code has to be an int. So the version name would be `123.45` but the version code will be `12345`